qlib/finco/workflow.py

```diff
@@ -123,30 +127,31 @@ def run(self, prompt: str) -> Path:
             self.set_context("user_prompt", prompt)
         self.logger.info(f"user_prompt: {self.get_context().get_context('user_prompt')}", title="Start")
 
-        # NOTE: list may not be enough for general task list
-        task_list = [WorkflowTask(), RecorderTask(), SummarizeTask()]
-        task_finished = []
-        while len(task_list):
-            task_list_info = [str(task) for task in task_list]
-
-            # task list is not long, so sort it is not a big problem
-            # TODO: sort the task list based on the priority of the task
-            # task_list = sorted(task_list, key=lambda x: x.task_type)
-            t = task_list.pop(0)
-            self.logger.info(f"Task finished: {[str(task) for task in task_finished]}",
-                             f"Task in queue: {task_list_info}",
-                             f"Executing task: {str(t)}",
-                             title="Task")
-
-            t.assign_context_manager(self._context)
-            res = t.execute()
-            t.summarize()
-            task_finished.append(t)
-            self.logger.plain_info(f"{str(t)} finished.\n\n\n")
-
-            for _ in res:
-                if not isinstance(t, (WorkflowTask, PlanTask, ActionTask, RecorderTask, AnalysisTask, SummarizeTask)):
-                    raise NotImplementedError(f"Unsupported Task type {_}")
-            task_list = res + task_list
-
+        # todo: add early stop condition
+        for i in range(10):
```

Comment:
I don't think the learning process should be coupled with Workflow Manager.

Reply:
Added learnManager